Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -21,6 +21,145 @@
 using namespace ablastr::fields;
 using warpx::fields::FieldType;
 
+namespace
+{
+
+    /**
+    * \brief Compute the minimal area for stability for the face i, j, k with normal 'dim'.
+    *
+    * \tparam dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+    *
+    * \param[in] i, j, k the indices of the cell
+    * \param[in] lx, ly, lz arrays containing the edge lengths
+    * \param[in] dx, dy, dz the mesh with in each direction
+    */
+    template <int dim>
+    [[nodiscard]]
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    amrex::Real
+    ComputeSStab(const int i, const int j, const int k,
+             const amrex::Array4<const amrex::Real> lx,
+             const amrex::Array4<const amrex::Real> ly,
+             const amrex::Array4<const amrex::Real> lz,
+             const amrex::Real dx, const amrex::Real dy, const amrex::Real dz)
```

Comment:
Should we leave a whitespace between the function name and the bracket and align all arguments?
```suggestion
    ComputeSStab (const int i, const int j, const int k,
                  const amrex::Array4<const amrex::Real> lx,
                  const amrex::Array4<const amrex::Real> ly,
                  const amrex::Array4<const amrex::Real> lz,
                  const amrex::Real dx, const amrex::Real dy, const amrex::Real dz)
```